packages/signers/src/magic/signer.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  WalletClientSigner,
+  type AuthSmartAccountSigner,
+  type SignTypedDataParams,
+} from "@alchemy/aa-core";
+import { Magic, type MagicUserMetadata } from "magic-sdk";
+import { createWalletClient, custom, type Hash } from "viem";
+import type { MagicAuthParams } from "./types.js";
+
+export class MagicSigner
+  implements AuthSmartAccountSigner<Magic, MagicAuthParams, MagicUserMetadata>
+{
+  inner: Magic;
+  private signer: WalletClientSigner | undefined;
+
+  constructor({ inner }: { inner: Magic }) {
```

Comment:
I think the constructor params here should be like magic SDK init params rather than a magic sdk directly. That might be a nicer user experience

Reply:
because `inner` is public you can still use it to create your auth params eg `authenticate: () => signer.inner.connectWithUI`

Reply:
hmm would that not be dependency injection? I imagine it's easier to test + less shared context for the Signer if the Magic client is just passed in.

Reply:
yea that's true, this wouldn't follow dependency injection. but it creates a potentially better devex

we can still test because we can mock whole modules, so we can always mock what `Magic` is anyways